src/applications/personalization/profile/tests/components/direct-deposit/UpdateErrorAlert.unit.spec.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { expect } from 'chai';
+import { render } from '@testing-library/react';
+
+import mockDisabilityCompensations from '@@profile/mocks/endpoints/disability-compensations';
+
+const { errors } = mockDisabilityCompensations.updates;
+
+import { UpdateErrorAlert } from '../../../components/direct-deposit/alerts/UpdateErrorAlert';
+
+describe('<UpdateErrorAlert />', () => {
+  it('renders', () => {
+    const { container } = render(
+      <UpdateErrorAlert saveError={errors.generic} />,
+    );
+    expect(container.innerHTML).to.not.be.empty;
+  });
+
+  it('renders the default error when it gets an unrecognized error message', async () => {
+    const { findByText } = render(
+      <UpdateErrorAlert saveError={errors.generic.errors} />,
+    );
+    expect(
+      await findByText(
+        'We’re sorry. We couldn’t update your payment information. Please try again later.',
+      ),
+    ).to.exist;
+  });
+
+  it('renders the invalid routing number error for checksum error', async () => {
+    const { findByText } = render(
+      <UpdateErrorAlert
+        saveError={errors.invalidChecksumRoutingNumber.errors}
+      />,
+    );
+    expect(
+      await findByText(
+        'We can’t find a bank linked to the routing number you entered.',
+      ),
+    ).to.exist;
+  });
+
+  it('renders the invalid routing number error for invalid error', async () => {
+    const { findByText: findByText2 } = render(
+      <UpdateErrorAlert saveError={errors.invalidRoutingNumber.errors} />,
+    );
+    expect(
+      await findByText2(
+        'We can’t find a bank linked to the routing number you entered.',
+      ),
+    ).to.exist;
+  });
+
+  it('renders the flagged account error', async () => {
+    const { findByText } = render(
+      <UpdateErrorAlert saveError={errors.accountNumberFlagged.errors} />,
+    );
+    expect(
+      await findByText(
+        'We’re sorry. You can’t change your direct deposit information right now because we’ve locked the ability to edit this information. We do this to protect your bank account information and prevent fraud when we think there may be a security issue.',
```

Comment:
I go back an forth on this, its a good callout. "Best practices" often times say that testing the actual things the user sees like text or content is a good thing to do and verifies user experience, but it can make a test harder to maintain when the text changes.

Per the testing library docs:

> In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names.

If it was something I was going to expect to change often I might change it to use a testid.

Reply:
thanks! good points. i think i usually avoid long copy for tests since if they change, we also need to change the test, and just adds more copy-pasta :) but you're right, snce these are just paraagraphs, we'd probably have o add some unnecessary attribute to the DOM n order o test another way 🤔